plugin/trino-ranger/src/main/java/io/trino/plugin/ranger/RangerSystemAccessControlImpl.java

```diff
@@ -0,0 +1,1005 @@
+        RangerAccessResult result = getRowFilterResult(request);
+
+        ViewExpression viewExpression = null;
+        if (isRowFilterEnabled(result)) {
+            String filter = result.getFilterExpr();
+
+            viewExpression = new ViewExpression(
+                    context.getIdentity().getUser(),
+                    Optional.of(tableName.getCatalogName()),
+                    Optional.of(tableName.getSchemaTableName().getSchemaName()),
+                    filter);
+        }
+
+        return List.of(viewExpression);
+    }
+
+    @Override
+    public List<ViewExpression> getColumnMasks(SystemSecurityContext context, CatalogSchemaTableName tableName, String columnName, Type type)
+    {
+        RangerTrinoAccessRequest request = createAccessRequest(
+                createResource(tableName.getCatalogName(), tableName.getSchemaTableName().getSchemaName(),
+                        tableName.getSchemaTableName().getTableName(), Optional.of(columnName)),
+                context, TrinoAccessType.SELECT);
+        RangerAccessResult result = getDataMaskResult(request);
+
+        ViewExpression viewExpression = null;
+        if (isDataMaskEnabled(result)) {
+            String maskType = result.getMaskType();
+            RangerServiceDef.RangerDataMaskTypeDef maskTypeDef = result.getMaskTypeDef();
+            String transformer = null;
+
+            if (maskTypeDef != null) {
+                transformer = maskTypeDef.getTransformer();
+            }
+
+            if (StringUtils.equalsIgnoreCase(maskType, RangerPolicy.MASK_TYPE_NULL)) {
+                transformer = "NULL";
+            }
+            else if (StringUtils.equalsIgnoreCase(maskType, RangerPolicy.MASK_TYPE_CUSTOM)) {
+                String maskedValue = result.getMaskedValue();
+
+                if (maskedValue == null) {
+                    transformer = "NULL";
+                }
+                else {
+                    transformer = maskedValue;
+                }
+            }
+
+            if (StringUtils.isNotEmpty(transformer)) {
+                transformer = transformer.replace("{col}", columnName).replace("{type}", type.getDisplayName());
+            }
+
+            viewExpression = new ViewExpression(
+                    context.getIdentity().getUser(),
+                    Optional.of(tableName.getCatalogName()),
+                    Optional.of(tableName.getSchemaTableName().getSchemaName()),
+                    transformer);
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("getColumnMask: user: %s, catalog: %s, schema: %s, transformer: %s");
+            }
+        }
+
+        return List.of(viewExpression);
+    }
+
+    @Override
+    public Set<String> filterCatalogs(SystemSecurityContext context, Set<String> catalogs)
+    {
+        LOG.debug("==> RangerSystemAccessControl.filterCatalogs(" + catalogs + ")");
```

Comment:
use `io.trino.plugin.base.util.LoggingInvocationHandler` instead of this logging.

Reply:
@dprophet

Reply:
Do you have an example of the LoggingInvocationHandler? There is no usage nor examples in Trino code base. The standard in all the code is io.airlift.log.Logger.